File: Make Human Unity/Assets/MakeHuman/Icons/KsSoft/Editor/Multilingual/tools/protocol/t_WinProperty.py
from CVariable import CReadVariable,CWriteVariable
from Vector import *
#/==========================================================================
#*!
#	@brief	t_WinProperty
#/
class t_WinProperty:
	#/==========================================================================
	#*!
	#	@brief	Member
	#/
	m_value = 0	#S32
	#/==========================================================================
	#*!
	#	@brief	Constructor
	#/
	def __init__(self):
		self.clear()

	#/==========================================================================
	#*!
	#	@brief	Accessor
	#/
	def clear(self):
		self.m_value = 0

	def read(self,cVariable):
		self.m_value = cVariable.getS32()
		return True

	def write(self,cVariable):
		cVariable.putS32(self.m_value)
		return True

File: Make Human Unity/Assets/MakeHuman/Icons/KsSoft/Plugins/Protocol/t_WinProperty.cs
//==============================================================================================
/*!
	@file  KsSoft Protocol
	
	(counter SJIS string 京)
*/
//==============================================================================================
using UnityEngine;
using System.Collections;
using System.Collections.Generic;

using KS;
//==========================================================================
/*!
	@brief	t_WinProperty
*/
public class t_WinProperty : ISerializable {
	//==========================================================================
	/*!
		@brief	Member
	*/
	public int	m_value;
	//==========================================================================
	/*!
		@brief	Constructor
	*/
	public t_WinProperty() {
		clear();
	}
	//==========================================================================
	/*!
		@brief	Accessor
	*/
	public void clear() {
		m_value = 0;
	}
	public bool read(CReadVariable cVariable) {
		try {
			m_value = cVariable.getS32();
		} catch (System.Exception e) {
			Debug.LogError(e);return false;
		}
		return true;
	}
	public bool write(CWriteVariable cVariable) {
		cVariable.put(m_value);
		return true;
	}
};

